[Bug]:  Can not run test unless `checkJs`: false

### Checklist

- [does not exist] I have read and agree to the [Code of Conduct](code-of-conduct)
- [X] I have searched for [related issues](https://github.com/CoderLine/mocha-vscode/issues) and [discussions](https://github.com/CoderLine/mocha-vscode/discussions) but they did not match this problem.
- [X] I want to contribute a bugfix myself.

### Actual behavior

Node v18.12.0

.mocharc.js 
```json
{
  "require": [
    "ts-node/register"
  ],
  "node-option": [
    "experimental-specifier-resolution=node",
    "loader=ts-node/esm",
    "no-warnings"
  ],
  "extensions": [
    "ts"
  ],
  "spec": [
    "**/test/**/*.spec.ts",
    "**/docs/**/*.spec.ts"
  ],
  "watch-files": [
    "src",
    "modules",
    "examples"
  ],
  "ignore": [
    "**/node_modules/**",
    "**/dist/**"
  ],
  "timeout": "10000"
}
```
tsconfig.json
```json
{
	"compilerOptions": {
	"strict": true,
        "outDir": "dist",
        "allowJs": true,
        "checkJs": true,
        "target": "ES2022",
        "module": "ES2022"
	}
}
```
Get output 

```
✖ ERROR: TypeError: Could not load reporter "/Users/admin/git/mocha-vscode/out/reporter/fullJsonStreamReporter.js":

 ../mocha-vscode/out/reporter/fullJsonStreamReporter.js(7,20): error TS7006: Parameter 'to' implicitly has an 'any' type.
 ```

### Expected behavior

Test are running successfully 

### Minimal, Reproducible Example

Use provided mocharc.json and tsconfig.json and try to debug or run a test

### Output

`
✖ ERROR: TypeError: Could not load reporter "/Users/admin/git/mocha-vscode/out/reporter/fullJsonStreamReporter.js":

 ../mocha-vscode/out/reporter/fullJsonStreamReporter.js(7,20): error TS7006: Parameter 'to' implicitly has an 'any' type.
 `

### Plugin Version Details

131c2c66f099991c98bfdb76fd09e0b996abde66

### VS Code Version Details

1.87.0
019f4d1419fbc8219a181fab7892ebccf7ee29a2
arm64

### Further details

This might be a configuration issue but I have not yet found good options to circumvent this. Also tried to recompile this extensions with typings for the reporter but that did not help either

--- a/.esbuild.js
+++ b/.esbuild.js
@@ -1,23 +1,51 @@
-const esbuild = require("esbuild");
+const esbuild = require('esbuild');
+const fs = require('fs');
+const path = require('path');
 
-const watch = process.argv.includes("--watch");
-const minify = watch ? process.argv.includes("--minify") : !process.argv.includes("--no-minify");
+const watch = process.argv.includes('--watch');
+const minify = watch ? process.argv.includes('--minify') : !process.argv.includes('--no-minify');
 
 const ctx = esbuild.context({
-  entryPoints: ["src/extension.ts", "src/reporter/fullJsonStreamReporter.ts"],
-  tsconfig: "./tsconfig.json",
+  entryPoints: ['src/extension.ts', 'src/reporter/fullJsonStreamReporter.ts'],
+  tsconfig: './tsconfig.json',
   bundle: true,
-  external: ["vscode", "pnpapi", "mocha", "esbuild"],
+  external: ['vscode', 'pnpapi', 'mocha', 'esbuild'],
   sourcemap: !minify,
   minify,
-  platform: "node",
-  outdir: "out",
-  keepNames: !minify
+  platform: 'node',
+  outdir: 'out',
+  keepNames: !minify,
+  plugins: [
+    {
+      name: 'mocha-vscode-ts-nocheck',
+      async setup(build) {
+
+        const write = build.initialOptions.write;
+        build.initialOptions.write = false;
+
+        build.onEnd(async (result) => {
+          if(result.outputFiles && write === undefined || write) {
+            result.outputFiles.forEach(file => {
+              fs.mkdirSync(path.dirname(file.path), { recursive: true });
+              const fd = fs.openSync(file.path, 'w');
+              try {
+                fs.writeSync(fd, '// @ts-nocheck\n');
+                fs.writeSync(fd, file.contents);
+              }
+              finally {
+                fs.closeSync(fd);
+              }
+            });
+          }
+        });
+      },
+    },
+  ],
 });
 
 ctx
   .then((ctx) => (watch ? ctx.watch() : ctx.rebuild()))
   .then(
     () => !watch && process.exit(0),
-    () => process.exit(1)
+    () => process.exit(1),
   );

--- a/package.json
+++ b/package.json
@@ -150,6 +150,7 @@
     "prettier-eslint-cli": "^8.0.1",
     "prettier-plugin-organize-imports": "^3.2.4",
     "sinon": "^17.0.1",
+    "ts-node": "^10.9.2",
     "tsx": "^4.7.2",
     "typescript": "^5.4.4"
   },

--- a/src/configurationFile.ts
+++ b/src/configurationFile.ts
@@ -12,9 +12,9 @@ import * as fs from 'fs';
 import { minimatch } from 'minimatch';
 import * as path from 'path';
 import * as vscode from 'vscode';
-import which from 'which';
 import { DisposableStore } from './disposable';
 import { HumanError } from './errors';
+import { getPathToNode } from './node';
 
 type OptionsModule = {
   loadOptions(): IResolvedConfiguration;
@@ -38,7 +38,6 @@ export class ConfigurationFile implements vscode.Disposable {
   private _resolver?: resolveModule.Resolver;
   private _optionsModule?: OptionsModule;
   private _configModule?: ConfigModule;
-  private _pathToNode?: string;
   private _pathToMocha?: string;
 
   /** Cached read promise, invalided on file change. */
@@ -93,28 +92,11 @@ export class ConfigurationFile implements vscode.Disposable {
     this.readPromise = undefined;
   }
 
-  async getPathToNode() {
-    // We cannot use process.execPath as this points to code.exe which is an electron application
-    // also with ELECTRON_RUN_AS_NODE this can lead to errors (e.g. with the --import option)
-    // we prefer to use the system level node
-    if (!this._pathToNode) {
-      this.logChannel.debug('Resolving Node.js executable');
-      this._pathToNode = await which('node', { nothrow: true });
-      if (this._pathToNode) {
-        this.logChannel.debug(`Found Node.js in PATH at '${this._pathToNode}'`);
-      } else {
-        this._pathToNode = process.execPath;
-        this.logChannel.debug(`Node.js not found in PATH using '${this._pathToNode}' as fallback`);
-      }
-    }
-    return this._pathToNode;
-  }
-
   async getMochaSpawnArgs(customArgs: readonly string[]): Promise<string[]> {
     this._pathToMocha ??= await this._resolveLocalMochaPath('/bin/mocha.js');
 
     return [
-      await this.getPathToNode(),
+      await getPathToNode(this.logChannel),
       this._pathToMocha,
       '--config',
       this.uri.fsPath,

--- a/src/extension.ts
+++ b/src/extension.ts
@@ -15,6 +15,7 @@ import * as vscode from 'vscode';
 import { ConfigValue } from './configValue';
 import { configFilePattern, getControllersForTestCommand } from './constants';
 import { Controller } from './controller';
+import { getPathToNode } from './node';
 import { TestRunner } from './runner';
 import { SourceMapStore } from './source-map-store';
 
@@ -113,13 +114,15 @@ export function deactivate() {}
 async function initESBuild(context: vscode.ExtensionContext, logChannel: vscode.LogOutputChannel) {
   logChannel.debug('Installing ESBuild binary');
 
+  const node = await getPathToNode(logChannel);
   const cli = await new Promise<ChildProcessWithoutNullStreams>((resolve, reject) => {
-    const p = spawn(process.execPath, ['install.js'], {
+    const p = spawn(node, ['install.js'], {
       cwd: path.join(context.extensionPath, 'node_modules', 'esbuild'),
       env: {
         ...process.env,
         ELECTRON_RUN_AS_NODE: '1',
       },
+      windowsHide: true,
     });
     p.on('spawn', () => resolve(p));
     p.on('error', reject);

--- a/src/node.ts
+++ b/src/node.ts
@@ -1,0 +1,30 @@
+/**
+ * Copyright (C) Daniel Kuschny (Danielku15) and contributors.
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ */
+
+import * as vscode from 'vscode';
+import which from 'which';
+
+let pathToNode: string | undefined = undefined;
+
+export async function getPathToNode(logChannel: vscode.LogOutputChannel) {
+  // We cannot use process.execPath as this points to code.exe which is an electron application
+  // also with ELECTRON_RUN_AS_NODE this can lead to errors (e.g. with the --import option)
+  // we prefer to use the system level node
+  if (!pathToNode) {
+    logChannel.debug('Resolving Node.js executable');
+    pathToNode = await which('node', { nothrow: true });
+    if (pathToNode) {
+      logChannel.debug(`Found Node.js in PATH at '${pathToNode}'`);
+    } else {
+      pathToNode = process.execPath;
+      logChannel.debug(`Node.js not found in PATH using '${pathToNode}' as fallback`);
+    }
+  }
+  return pathToNode;
+}

--- a/src/test/integration/tsnode-checkjs.test.ts
+++ b/src/test/integration/tsnode-checkjs.test.ts
@@ -1,0 +1,38 @@
+/**
+ * Copyright (C) Daniel Kuschny (Danielku15) and contributors.
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ */
+
+import { expect } from 'chai';
+import * as vscode from 'vscode';
+import { captureTestRun, expectTestTree, getController } from '../util';
+
+describe('tsnode-checkjs', () => {
+  it('discovers tests', async () => {
+    const c = await getController();
+    await expectTestTree(c, [['test', [['hello.spec.ts', [['math', [['addition']]]]]]]]);
+  });
+
+  it('runs tests', async () => {
+    const c = await getController();
+    const profiles = c.profiles;
+    expect(profiles).to.have.lengthOf(2);
+
+    const run = await captureTestRun(
+      c,
+      new vscode.TestRunRequest(
+        undefined,
+        undefined,
+        profiles.find((p) => p.kind === vscode.TestRunProfileKind.Run),
+      ),
+    );
+
+    run.expectStates({
+      'test/hello.spec.ts/math/addition': ['enqueued', 'started', 'passed'],
+    });
+  });
+});

--- a/test-workspaces/tsnode-checkjs/.mocharc.json
+++ b/test-workspaces/tsnode-checkjs/.mocharc.json
@@ -1,0 +1,27 @@
+{
+    "require": [
+      "ts-node/register"
+    ],
+    "node-option": [
+      "experimental-specifier-resolution=node",
+      "loader=ts-node/esm",
+      "no-warnings"
+    ],
+    "extensions": [
+      "ts"
+    ],
+    "spec": [
+      "**/test/**/*.spec.ts"
+    ],
+    "watch-files": [
+      "src",
+      "modules",
+      "examples"
+    ],
+    "ignore": [
+      "**/node_modules/**",
+      "**/dist/**"
+    ],
+    "timeout": "10000"
+  }
+  

--- a/test-workspaces/tsnode-checkjs/package.json
+++ b/test-workspaces/tsnode-checkjs/package.json
@@ -1,0 +1,4 @@
+{
+    "name": "tsnode-checkjs",
+    "type": "module"
+}

--- a/test-workspaces/tsnode-checkjs/test/hello.spec.ts
+++ b/test-workspaces/tsnode-checkjs/test/hello.spec.ts
@@ -1,0 +1,7 @@
+import { strictEqual } from 'node:assert';
+
+describe('math', () => {
+  it('addition', async () => {
+    strictEqual((1 + 1) as number, 2 as any as number);
+  });
+});

--- a/test-workspaces/tsnode-checkjs/tsconfig.json
+++ b/test-workspaces/tsnode-checkjs/tsconfig.json
@@ -1,0 +1,10 @@
+{
+  "compilerOptions": {
+    "strict": true,
+    "outDir": "dist",
+    "allowJs": true,
+    "checkJs": true,
+    "target": "ES2022",
+    "module": "ES2022"
+  }
+}

